src/main/java/org/opentripplanner/middleware/persistence/TypedPersistence.java

```diff
@@ -134,53 +135,76 @@ public T getById(String id) {
     }
 
+    /**
+     * Build a filter for querying Mongo based on userId and from/to dates.
+     */
+    public static Bson filterByUserAndDateRange(String userId, Date fromDate, Date toDate) {
+        Set<Bson> clauses = new HashSet<>();
+        if (userId == null) {
+            throw new IllegalArgumentException("userId is required to be non-null.");
+        }
+        // user id is required, so as a minimum return all entities for user.
+        clauses.add(eq("userId", userId));
+        // Get all entities created since the supplied "from date".
+        if (fromDate != null) {
+            clauses.add(gte("dateCreated", fromDate));
+        }
+        // Get all entities created until the supplied "to date".
+        if (toDate != null) {
+            clauses.add(lte("dateCreated", toDate));
+        }
+        return Filters.and(clauses);
+    }
+
+    /**
+     * Build a filter for querying Mongo based on userId.
+     */
+    public static Bson filterByUserId(String userId) {
+        return filterByUserAndDateRange(userId, null, null);
+    }
+
     /**
      * Return the number of items based on the supplied Mongo filter
-     *
      */
     public long getCountFiltered(Bson filter) {
         return mongoCollection.countDocuments(filter);
     }
 
     /**
-     * Get all objects satisfying the supplied Mongo filter.
-     * This ties our persistence directly to Mongo for now but is expedient.
-     * We should really have a bit more abstraction here.
+     * Get all objects satisfying the supplied Mongo filter. This ties our persistence directly to Mongo for now but is
+     * expedient. We should really have a bit more abstraction here.
      */
     public List<T> getFiltered(Bson filter) {
         return mongoCollection.find(filter).into(new ArrayList<>());
     }
 
     /**
-     * Expose the internal MongoCollection to the caller.
-     * This ties our persistence directly to Mongo for now but is expedient.
-     * We will write all the queries we need in the calling methods, then make an abstraction here on TypedPersistence
-     * once we see everything we need to support.
+     * Expose the internal MongoCollection to the caller. This ties our persistence directly to Mongo for now but is
+     * expedient. We will write all the queries we need in the calling methods, then make an abstraction here on
+     * TypedPersistence once we see everything we need to support.
      */
     public MongoCollection<T> getMongoCollection() {
         return this.mongoCollection;
     }
 
     /**
-     * Get all objects satisfying the supplied Mongo filter.
-     * This ties our persistence directly to Mongo for now but is expedient.
-     * We should really have a bit more abstraction here.
+     * Get all objects satisfying the supplied Mongo filter. This ties our persistence directly to Mongo for now but is
+     * expedient. We should really have a bit more abstraction here.
      */
```

Comment:
There are a lot of formatting changes just to comments, we should try to avoid this because it clutters up PRs.